virttest/vt_utils/cpu.py

```diff
@@ -0,0 +1,131 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat (c) 2023
+# Author: Houqi Zuo <hzuo@redhat.com>
+import re
+
+from avocado.utils import process
+
+
+def get_cpu_info():
+    """
+    Return information about the CPU information.
+
+    :return: cpu information
+    :rtype: list[dict]
+    """
+    cpu_info = list()
+    with open("/proc/cpuinfo") as fd:
+        processors = fd.read().split("\n\n")
+    for processor in processors:
+        if processor != "":
+            info = dict(map(lambda x: [i.strip() for i in x.split(":", 1)],
+                            processor.split("\n")))
+            cpu_info.append(info)
+
+    return cpu_info
+
+
+def get_cpu_model_name():
+    """
+    Return physical cpu model.
+
+    :return: cpu model name
+    :rtype: string
+    :raises: An OSError will be raised if it's NOT available on the platform
+    """
+    cpu_model = process.run("lscpu").stdout_text
+    cpu_model_re = "[mM]odel name:.*\n"
+    cpu_model = re.findall(cpu_model_re, cpu_model)
+    if not cpu_model:
+        raise OSError("The cpu model name was NOT found!")
+    return cpu_model[0].split(":", 1)[-1].strip()
+
+
+def get_cpu_flags():
+    """
+    Return a list of the CPU flags.
+
+    :return: cpu flags
+    :rtype: list[string]
+    :raises: An OSError will be raised if it's NOT available on the platform
+    """
+    cpu_flags_re = "flags\s+:\s+([\w\s]+)\n"
+    with open("/proc/cpuinfo") as fd:
+        cpu_info = fd.read()
+    cpu_flag_lists = re.findall(cpu_flags_re, cpu_info)
+    if not cpu_flag_lists:
+        raise OSError("The cpu flags were NOT found!")
+    cpu_flags = cpu_flag_lists[0]
+    return re.split("\s+", cpu_flags.strip())
+
+
+def get_cpu_features():
+    """
+    Return a list of the CPU features.
+
+    :return: cpu features
+    :rtype: list[string]
+    :raises: An OSError will be raised if it's NOT available on the platform
+    """
+    cpu_features_re = "[fF]eatures\s+:\s+([\w\s]+)\n"
+    with open("/proc/cpuinfo") as fd:
+        cpu_info = fd.read()
+    cpu_features_lists = re.findall(cpu_features_re, cpu_info)
+    if not cpu_features_lists:
+        raise OSError("The cpu features were NOT found!")
+    cpu_features = cpu_features_lists[0]
+    return re.split("\s+", cpu_features.strip())
+
+
+def get_cpu_vendor_id():
+    """
+    Return the name of the CPU vendor ID.
+
+    :return: the name of the CPU vendor ID in string
+    :rtype: string
+    """
+    vendor_re = "vendor_id\s+:\s+(\w+)"
+    with open("/proc/cpuinfo") as fd:
+        cpu_info = fd.read()
+    vendor_id = re.findall(vendor_re, cpu_info)
+    return vendor_id[0]
+
+
+def get_cpu_family():
+    """
+    Return the name of cpu family.
+
+    :return: the name of the cpu family in string
+    :rtype: string
+    """
+    cpu_family_re = "cpu family\s+:\s+(\w+)"
```

Comment:
do you think 'c' in 'cpu family' could be a capital letter?

Please also keep an eye on stepping

Reply:
> do you think 'c' in 'cpu family' could be a capital letter?
> 
> Please also keep an eye on stepping

After I made check, the 'C' is NOT necessary at here since the x86_64 platform only has `cpu family`.